src/core/util/log.cc

```diff
@@ -46,8 +48,82 @@ BOOST_LOG_GLOBAL_LOGGER_INIT(g_Logger, boost::log::sources::severity_logger_mt)
+{
+  std::stringstream ss;
+  ss << "Request source : \""
+     << static_cast<std::string>(boost::network::http::source(req))
+     << "\" dest : \""
+     << static_cast<std::string>(boost::network::http::destination(req))
+     << "\" headers: ";
+  for (auto const& header : req.headers())
+    ss << " <" << header.first << ">  : \"" << header.second << "\"";
+  std::string body(static_cast<std::string>(boost::network::http::body(req)));
+  ss << " body: \""
+     << kovri::core::GetFormattedHex(
+            reinterpret_cast<const uint8_t*>(body.data()), body.length())
+     << "\"";
+  return ss.str();
+}
+
+std::string LogNetResponseToString(
+    const boost::network::http::client::response& response)
+{
+  std::stringstream ss;
+  ss << "Response source: \""
+     << static_cast<std::string>(boost::network::http::source(response))
+     << "\" dest: \""
+     << static_cast<std::string>(boost::network::http::destination(response))
+     << "\" headers: ";
+  for (auto const& header : response.headers())
+    ss << " <" << header.first << ">  : \"" << header.second << "\"";
+
+  std::string body(
+      static_cast<std::string>(boost::network::http::body(response)));
+  ss << " body: "
+     << kovri::core::GetFormattedHex(
+            reinterpret_cast<const uint8_t*>(body.data()), body.length());
+  ss << " version: \""
+     << static_cast<std::string>(boost::network::http::version(response))
+     << "\" status: \""
+     << static_cast<std::string>(boost::network::http::status_message(response))
+     << "\"";
+  return ss.str();
+}
+
+std::string LogNetOptionsToString(
+    const boost::network::http::client::options& opt)
+{
+  std::stringstream ss;
+  ss << "Options : cache_resolved ? " << opt.cache_resolved()
+     << ", follow_redirects ? " << opt.follow_redirects() << ", certificate ? "
+     << (opt.openssl_certificate() ? *opt.openssl_certificate()
+                                   : std::string("No"))
+     << ", openssl_certificate_file ? "
+     << (opt.openssl_certificate_file() ? *opt.openssl_certificate_file()
+                                        : std::string("No"))
+     << ", openssl_private_key_file ? "
+     << (opt.openssl_private_key_file() ? *opt.openssl_private_key_file()
+                                        : std::string("No"))
+     << ", verify_path ? "
+     << (opt.openssl_verify_path() ? *opt.openssl_verify_path()
+                                   : std::string("No"))
+     << ", openssl_ciphers ? "
+     << (opt.openssl_ciphers() ? *opt.openssl_ciphers() : std::string("No"))
+     << ", openssl_sni_hostname ? "
+     << (opt.openssl_sni_hostname() ? *opt.openssl_sni_hostname()
+                                    : std::string("No"))
+     << ", always_verify_peer ? " << opt.always_verify_peer()
+     << ", options : " << opt.openssl_options()
+     << ", timeout : " << opt.timeout() << ", io_service ? "
+     << (opt.io_service() ? "Yes" : "No");
+  return ss.str();
+}
```

Comment:
This is a lot of manual work; especially if we add/remove options. I'm not sure how useful logging these options are right now but if this is what you want then ok.

As a whole, I think we should also consider a more unified system of formatting. All these dangling `?` and `<>` and `:` in terms of English, I find quite confusing.